### PR TITLE
Select available pool when create volume

### LINKED ISSUE #1179: Profile should not select the unavailable pool
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
When a pool has status "unavailable", it should not be used to create volume by profile.

**What you expected to happen**:
Profile should not use the unavailable pool to create volume

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ThisIsClark**
**What this PR does / why we need it**:
Select a available pool when create volume

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1179

**Special notes for your reviewer**:

**Release note**:

```release-note
[root@master-node opensds]# osdsctl pool list
+--------------------------------------+-----------------+-------------+-------------+---------------+--------------+
| Id                                   | Name            | Description | Status      | TotalCapacity | FreeCapacity |
+--------------------------------------+-----------------+-------------+-------------+---------------+--------------+
| e94ebaca-76be-56c7-aa12-b348632cc30f | StoragePool002  |             | unavailable | 40            | 36           |
+--------------------------------------+-----------------+-------------+-------------+---------------+--------------+
[root@master-node opensds]# osdsctl volume create 1 --name=test-001
+------------------+--------------------------------------+
| Property         | Value                                |
+------------------+--------------------------------------+
| Id               | 3162a008-f1f1-4e14-b270-f9a0d80d53bc |
| CreatedAt        | 2020-01-01T12:09:44                  |
| Name             | test-001                             |
| Description      |                                      |
| GroupId          |                                      |
| Size             | 1                                    |
| AvailabilityZone | default                              |
| Status           | creating                             |
| PoolId           |                                      |
| ProfileId        | 85bf0d5b-f0ed-4573-ac1d-a21d61402bf9 |
| Metadata         | null                                 |
|                  |                                      |
| MultiAttach      | false                                |
+------------------+--------------------------------------+
[root@master-node opensds]# osdsctl volume list
+--------------------------------------+----------+-------------+------+------------------+--------+--------------------------------------+
| Id                                   | Name     | Description | Size | AvailabilityZone | Status | ProfileId                            |
+--------------------------------------+----------+-------------+------+------------------+--------+--------------------------------------+
| 3162a008-f1f1-4e14-b270-f9a0d80d53bc | test-001 |             | 1    | default          | error  | 85bf0d5b-f0ed-4573-ac1d-a21d61402bf9 |
+--------------------------------------+----------+-------------+------+------------------+--------+--------------------------------------+
```

### CODE REVIEW

**Comment by kumarashit** [comment]
Hi @ThisIsClark , from the release notes, it's not clear what is the test result. Pool is unavailable, volume list doesn't have pool name and then the volume goes in error state. Can you please clarify more

**Comment by sushanthakumar** [comment]
> **What this PR does / why we need it**:
> Select a available pool when create volume
> 
> **Which issue this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)_: fixes #1179
> 
> **Special notes for your reviewer**:
> Any other PR(s) this PR is dependant on: None

> **Release note**:
> 
> ```
> [root@master-node opensds]# osdsctl pool list
> +--------------------------------------+-----------------+-------------+-------------+---------------+--------------+
> | Id                                   | Name            | Description | Status      | TotalCapacity | FreeCapacity |
> +--------------------------------------+-----------------+-------------+-------------+---------------+--------------+
> | e94ebaca-76be-56c7-aa12-b348632cc30f | StoragePool002  |             | unavailable | 40            | 36           |
> +--------------------------------------+-----------------+-------------+-------------+---------------+--------------+
> [root@master-node opensds]# osdsctl volume create 1 --name=test-001
> +------------------+--------------------------------------+
> | Property         | Value                                |
> +------------------+--------------------------------------+
> | Id               | 3162a008-f1f1-4e14-b270-f9a0d80d53bc |
> | CreatedAt        | 2020-01-01T12:09:44                  |
> | Name             | test-001                             |
> | Description      |                                      |
> | GroupId          |                                      |
> | Size             | 1                                    |
> | AvailabilityZone | default                              |
> | Status           | creating                             |
> | PoolId           |                                      |
> | ProfileId        | 85bf0d5b-f0ed-4573-ac1d-a21d61402bf9 |
> | Metadata         | null                                 |
> |                  |                                      |
> | MultiAttach      | false                                |
> +------------------+--------------------------------------+
> [root@master-node opensds]# osdsctl volume list
> +--------------------------------------+----------+-------------+------+------------------+--------+--------------------------------------+
> | Id                                   | Name     | Description | Size | AvailabilityZone | Status | ProfileId                            |
> +--------------------------------------+----------+-------------+------+------------------+--------+--------------------------------------+
> | 3162a008-f1f1-4e14-b270-f9a0d80d53bc | test-001 |             | 1    | default          | error  | 85bf0d5b-f0ed-4573-ac1d-a21d61402bf9 |
> +--------------------------------------+----------+-------------+------+------------------+--------+--------------------------------------+
> ```

**Comment by sushanthakumar** [comment]
Issue observed was after pool is detected and marked as available, when the backend connection is cutoff, pool is marked as unavailable. At this point of time, if we try to create a volume, the unavailble pool is picked for volume creation which seems incorrect. Any other thought on this flow or are we missing something here

**Comment by joseph-v** [comment]
LGTM

File: pkg/controller/selector/selector.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func (s *selector) SelectSupportedPoolForVolume(vol *model.VolumeSpec) (*model.S
 			filterRequest["id"] = vol.PoolId
 		}
 		filterRequest["storageType"] = prf.StorageType
+		filterRequest["status"] = "available"
```
Comment (NajmudheenCT) [inline pkg/controller/selector/selector.go]: Do we know what all the possible values for poll status? Adding this filter limits to creation only if status is set as "Available"

Comment (NajmudheenCT) [reply pkg/controller/selector/selector.go]: Asper discussion we have to have a standard enum for opensds pool status. as of now only available pools can use for volume creation

**Pull request merged**